Panic using fmt on filesystem path that does not exist

```
$ templ fmt "this path does not exist"
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xefe7626]

goroutine 36 [running]:
github.com/a-h/templ/cmd/templ/processor.Process.func1.FindTemplates.1({0x7ff7b21e99a2, 0x18}, {0x0, 0x0}, {0x0?, 0x0?})
        /Users/jkingsbury/go/pkg/mod/github.com/a-h/templ@v0.2.598/cmd/templ/processor/processor.go:46 +0x26
path/filepath.Walk({0x7ff7b21e99a2, 0x18}, 0xc000216fb8)
        /Users/jkingsbury/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-amd64/src/path/filepath/path.go:558 +0x4a
github.com/a-h/templ/cmd/templ/processor.FindTemplates(...)
        /Users/jkingsbury/go/pkg/mod/github.com/a-h/templ@v0.2.598/cmd/templ/processor/processor.go:45
github.com/a-h/templ/cmd/templ/processor.Process.func1()
        /Users/jkingsbury/go/pkg/mod/github.com/a-h/templ@v0.2.598/cmd/templ/processor/processor.go:22 +0x7a
created by github.com/a-h/templ/cmd/templ/processor.Process in goroutine 35
        /Users/jkingsbury/go/pkg/mod/github.com/a-h/templ@v0.2.598/cmd/templ/processor/processor.go:20 +0xa7
```

--- a/cmd/templ/processor/processor.go
+++ b/cmd/templ/processor/processor.go
@@ -43,6 +43,9 @@ func shouldSkipDir(dir string) bool {
 
 func FindTemplates(srcPath string, output chan<- string) (err error) {
 	return filepath.Walk(srcPath, func(currentPath string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 		if info.IsDir() && shouldSkipDir(currentPath) {
 			return filepath.SkipDir
 		}

--- a/cmd/templ/processor/processor_test.go
+++ b/cmd/templ/processor/processor_test.go
@@ -1,0 +1,19 @@
+package processor
+
+import (
+	"os"
+	"testing"
+)
+
+func TestFindTemplates(t *testing.T) {
+	t.Run("returns an error if the directory does not exist", func(t *testing.T) {
+		output := make(chan string)
+		err := FindTemplates("nonexistent", output)
+		if err == nil {
+			t.Fatal("expected error, but got nil")
+		}
+		if !os.IsNotExist(err) {
+			t.Fatalf("expected os.IsNotExist(err) to be true, but got: %v", err)
+		}
+	})
+}

